mmaction/core/evaluation/eval_hooks.py

```diff
@@ -123,6 +124,11 @@ def evaluate(self, runner, results):
             runner.log_buffer.output[name] = val
         runner.log_buffer.ready = True
         if self.key_indicator is not None:
+            if self.key_indicator not in eval_res:
+                warnings.warn('The key indicator for evaluation is not '
```

Comment:
should this be a warning or an error?

Reply:
A warning ? because this misalignment will not influence the evaluation and the users still can find the best score manually from the json file.

Reply:
where should I put key_indicator in config file. Is it like below
```
evaluation = dict(interval=1, metrics=['mean_average_precision'], key_indicator = 'mean_average_precision')
```

Reply:
> where should I put key_indicator in config file. Is it like below
> 
> ```
> evaluation = dict(interval=1, metrics=['mean_average_precision'], key_indicator = 'mean_average_precision')
> ```

Yep

Reply:
It is auc not mean_average_precision, right?

Reply:
You can assign `rule='greater'` and `key_indicator='mean_average_precision'`.